# Question 2

## step 1: load data into stata

In [1]:
use "/Users/ning/Dropbox/Working_Directory/Tex_MD/Teaching/tutorial/6002/W2/2021S2/econmath.dta", clear

## step 2: find the summary statistics for "actmth", "acteng" and "score"

In [2]:
sum actmth acteng score


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
      actmth |        814     23.2113    3.773354         12         36
      acteng |        814    22.59459    3.788735         12         34
       score |        856    72.59981    13.40068      19.53      98.44


## step 3: regression result of  $score = \beta_0 + \beta_1 \cdot colgpa + \beta_2 \cdot actmth + \beta_3 \cdot acteng + u$

In [3]:
reg score colgpa actmth acteng


      Source |       SS           df       MS      Number of obs   =       814
-------------+----------------------------------   F(3, 810)       =    177.94
       Model |  57165.5698         3  19055.1899   Prob > F        =    0.0000
    Residual |  86743.1975       810  107.090367   R-squared       =    0.3972
-------------+----------------------------------   Adj R-squared   =    0.3950
       Total |  143908.767       813  177.009554   Root MSE        =    10.348

------------------------------------------------------------------------------
       score |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      colgpa |    12.3662   .7150624    17.29   0.000     10.96261    13.76979
      actmth |   .8833519   .1121984     7.87   0.000      .663118    1.103586
      acteng |    .051764   .1110631     0.47   0.641    -.1662415    .2697695
       _cons |   16.17402   2.800439     5.78   0.

### $score = 16.17402 + 12.3662 \cdot colgpa + .8833519 \cdot actmth + .051764 \cdot acteng + u$
### $R^2 = 0.39$

### two regression inputs can be compared only if (1) they share the same unit (2) they share the similar standard deviation.

### in this case, $actmth$ and $acteng$ are in same unit (%) and with similar standard deviations. Hence, the one with a larger absolute value of the regression coefficient is the better one

### However, if variables are not in the same unit and their standard deviations are different, we need to "standardize them" before comparing them. By "standardizing", we minus each variable with its sample mean and divide each variable by its own standard deviation, or mathematically

### $$\frac{\mathbf{x} - \overline{\mathbf{x}}} {\mathrm{sd} \left( \mathbf{x} \right)}$$

## we standardize both inputs and outputs of regression, which gives us an interesting result as follows

In [4]:
egen mean_score  = mean(score)
egen mean_colgpa = mean(colgpa)
egen mean_actmth = mean(actmth)
egen mean_acteng = mean(acteng)

egen sd_score  = sd(score)
egen sd_colgpa = sd(colgpa)
egen sd_actmth = sd(actmth)
egen sd_acteng = sd(acteng)

gen score_st  = (score  - mean_score ) / sd_score
gen colgpa_st = (colgpa - mean_colgpa) / sd_colgpa
gen actmth_st = (actmth - mean_actmth) / sd_actmth
gen acteng_st = (acteng - mean_acteng) / sd_acteng

reg score_st colgpa_st actmth_st acteng_st












(42 missing values generated)

(42 missing values generated)


      Source |       SS           df       MS      Number of obs   =       814
-------------+----------------------------------   F(3, 810)       =    177.94
       Model |  318.332478         3  106.110826   Prob > F        =    0.0000
    Residual |  483.038603       810  .596343954   R-squared       =    0.3972
-------------+----------------------------------   Adj R-squared   =    0.3950
       Total |   801.37108       813  .985696286   Root MSE        =    .77223

------------------------------------------------------------------------------
    score_st |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
   colgpa_st |    .495423   .0286473    17.29   0.000     .4391913    .5516547
   actmth_st |   .2487336   .0315927     7.87   0.000     .1867203    .3107469
   acteng_st |   .0146351   .0314006     0.47   0.641   

### the coefficient of actmth_st ($0.2487336$) is the conditional correlation between score and actmth when you fix the value of acteng and colgpa. Larger the absolute value of the regression coefficient in standardized regression, more correlated the variable to $Y$ even though you control all other variables, hence better the input

# Question 3

In [5]:
use "/Users/ning/Dropbox/Working_Directory/Tex_MD/Teaching/tutorial/6002/W2/2021S2/hprice1.dta", clear

In [6]:
describe


Contains data from /Users/ning/Dropbox/Working_Directory/Tex_MD/Teaching/tutoria
> l/6002/W2/2021S2/hprice1.dta
  obs:            88                          
 vars:            10                          17 Mar 2002 12:21
 size:         2,816                          
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
price           float   %9.0g                 house price, $1000s
assess          float   %9.0g                 assessed value, $1000s
bdrms           byte    %9.0g                 number of bdrms
lotsize         float   %9.0g                 size of lot in square feet
sqrft           int     %9.0g                 size of house in square feet
colonial        byte    %9.0g                 =1 if home is colonial style
lprice          float   %9.0g                

## step 1: summary statistics

In [7]:
sum price sqrft bdrms


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
       price |         88     293.546    102.7134        111        725
       sqrft |         88    2013.693    577.1916       1171       3880
       bdrms |         88    3.568182    .8413926          2          7


## step 2: regression

In [8]:
reg price sqrft bdrms


      Source |       SS           df       MS      Number of obs   =        88
-------------+----------------------------------   F(2, 85)        =     72.96
       Model |  580009.152         2  290004.576   Prob > F        =    0.0000
    Residual |  337845.354        85  3974.65122   R-squared       =    0.6319
-------------+----------------------------------   Adj R-squared   =    0.6233
       Total |  917854.506        87  10550.0518   Root MSE        =    63.045

------------------------------------------------------------------------------
       price |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
       sqrft |   .1284362   .0138245     9.29   0.000     .1009495    .1559229
       bdrms |   15.19819   9.483517     1.60   0.113    -3.657582    34.05396
       _cons |    -19.315   31.04662    -0.62   0.536    -81.04399      42.414
--------------------------------------------------

In [9]:
egen mean_price = mean(price  )
egen mean_sqrft = mean(sqrft )
egen mean_bdrms = mean(bdrms)

egen sd_price = sd(price )
egen sd_sqrft = sd(sqrft )
egen sd_bdrms = sd(bdrms)

gen price_st = (price - mean_price) / sd_price
gen sqrft_st = (sqrft - mean_sqrft) / sd_sqrft
gen bdrms_st = (bdrms - mean_bdrms) / sd_bdrms

reg price_st sqrft_st bdrms_st












      Source |       SS           df       MS      Number of obs   =        88
-------------+----------------------------------   F(2, 85)        =     72.96
       Model |  54.9768962         2  27.4884481   Prob > F        =    0.0000
    Residual |  32.0230964        85  .376742311   R-squared       =    0.6319
-------------+----------------------------------   Adj R-squared   =    0.6233
       Total |  86.9999926        87  .999999915   Root MSE        =    .61379

------------------------------------------------------------------------------
    price_st |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
    sqrft_st |    .721739   .0776856     9.29   0.000     .5672791    .8761989
    bdrms_st |   .1244983   .0776856     1.60   0.113    -.0299616    .2789581
       _cons |   1.80e-07   .0654306     0.00   1.000    -.1300934    .1300937
----------------------------------------

## step 3: prediction on an "imaginary" observation

In [10]:
set obs `=_N+1'

gen id = _n

list price sqrft lotsize bdrms if id >= 88

replace sqrft  = 2438 if id == 89
replace bdrms  = 4    if id == 89

list price sqrft bdrms if id > 88

quietly reg price sqrft bdrms 
predict yhat, xb

list yhat price sqrft bdrms if id > 88


number of observations (_N) was 88, now 89



     +---------------------------------+
     | price   sqrft   lotsize   bdrms |
     |---------------------------------|
 88. |   242    1774      4950       4 |
 89. |     .       .         .       . |
     +---------------------------------+

(1 real change made)

(1 real change made)


     +-----------------------+
     | price   sqrft   bdrms |
     |-----------------------|
 89. |     .    2438       4 |
     +-----------------------+




     +----------------------------------+
     |     yhat   price   sqrft   bdrms |
     |----------------------------------|
 89. | 354.6053       .    2438       4 |
     +----------------------------------+


## step 4: given the real price of the "imaginary" observation is 300,000, compute the prediction residual (or prediction error) on an "imaginary" observation

In [11]:
replace price = 300 if id == 89

list yhat price sqrft bdrms if id > 88

gen residual = price - yhat
list residual yhat price sqrft bdrms in 89


(1 real change made)


     +----------------------------------+
     |     yhat   price   sqrft   bdrms |
     |----------------------------------|
 89. | 354.6053     300    2438       4 |
     +----------------------------------+



     +----------------------------------------------+
     |  residual       yhat   price   sqrft   bdrms |
     |----------------------------------------------|
 89. | -54.60526   354.6053     300    2438       4 |
     +----------------------------------------------+


In [13]:
!rm -rf W2_Stata.html
!jupyter nbconvert --to html W2_Stata.ipynb 





[NbConvertApp] Converting notebook W2_Stata.ipynb to html
[NbConvertApp] Writing 586618 bytes to W2_Stata.html
